vespamalloc/src/vespamalloc/malloc/allocchunk.h

```diff
@@ -20,16 +20,48 @@ struct TaggedPtr {
     TaggedPtr() noexcept : _ptr(nullptr), _tag(0) { }
     TaggedPtr(void *h, size_t t) noexcept : _ptr(h), _tag(t) {}
 
+#if defined(__x86_64__)
+#define VESPA_USE_ATOMIC_TAGGEDPTR
+    TaggedPtr load(std::memory_order = std::memory_order_seq_cst) {
```

Comment:
This reintroduces the ABA problem since we are not guaranteed that the load is atomic.

Reply:
Agree, I will do that.